atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java

```diff
@@ -76,6 +79,67 @@ public void testRenderJava8OptionalsWhenRequested() throws IOException {
                         containsString("import java.util.Optional")));
     }
 
+    @Test
+    public void testIgnoreTableNameLengthFlag() throws IOException {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        schema.ignoreTableNameLengthChecks();
+        String longTableName = String.join("", Collections.nCopies(1000, "x"));
+        TableReference tableRef = TableReference.createWithEmptyNamespace(longTableName);
+        schema.addTableDefinition(longTableName, getSimpleTableDefinition(tableRef));
+    }
+
+    @Test
+    public void testLongTableNameLengthFailsCassandra() throws IOException {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        int longLengthCassandra = AtlasDbConstants.CASSANDRA_TABLE_NAME_CHAR_LIMIT + 1;
+        String longTableName = String.join("", Collections.nCopies(longLengthCassandra, "x"));
+        TableReference tableRef = TableReference.createWithEmptyNamespace(longTableName);
+        assertThatThrownBy(() ->
+                schema.addTableDefinition(longTableName, getSimpleTableDefinition(tableRef)))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Internal table name %s is too long, exceeds Cassandra limit (%s). " +
```

Comment:
nit: We should refactor the message part, it can be a method like `getExceededName(String tableName, Enum dbType)`, similarly in the impl code too.